Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -251,7 +254,7 @@ std::optional<InteractionRegion> interactionRegionForRenderedRegion(RenderObject
         // Could be a `<label for="...">` or a label with a descendant.
         // In cases where both elements get a region we want to group them by the same `elementIdentifier`.
         auto associatedElement = downcast<HTMLLabelElement>(matchedElement)->control();
-        if (associatedElement) {
+        if (associatedElement && (!is<HTMLInputElement>(associatedElement) || !downcast<HTMLInputElement>(associatedElement)->isDisabledFormControl())) {
```

Comment:
```suggestion
        if (associatedElement && !associatedElement->isDisabledFormControl()) {
```

Nit - maybe a bit simpler if we omit the class check (unless this rule needs to only apply for `input` elements?)

Reply:
Thanks!